File: 667_Beautiful_Arrangement_II.cpp
#include<bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> constructArray(int n, int k) {
        /* O(n): 56% */
        int left = 1, right = n;
        vector<int> res;
        while (left <= right){
            if (k > 1 && k % 2){
                res.push_back(left++);
                k--;
            }
            else if (k > 1 && k % 2 == 0){
                res.push_back(right--); 
                k--;
            }
            else res.push_back(left++);
        }
        return res;
    }
};

int main(){
    Solution* solution = new Solution();
    auto res = solution->constructArray(5, 2);
    for (auto n : res) cout << n << " ";
    cout << "\n";
    return 0;
}

File: 667_Beautiful_Arrangement_II.py
class Solution(object):
    def constructArray1(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: List[int]
        """
        res = range(1, n + 1)
        for i in range(2, k + 1):
            res[i - 1:] = res[i - 1:][::-1]
        return res

    def constructArray2(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: List[int]
        """
        res = []
        i, j = 1, n
        while i <= j:
            if k > 1:
                if k % 2 == 0:
                    res.append(j)
                    j -= 1
                else:
                    res.append(i)
                    i += 1
                k -= 1
            else:
                res.append(i)
                i += 1
        return res
